.infrastructure/55_cognito.tf

```diff
@@ -169,6 +169,30 @@ resource "aws_cognito_user_pool" "devportal" {
     required                 = false
   }
 
+  schema {
+    name                     = "role"
+    attribute_data_type      = "String"
+    developer_only_attribute = false
+    mutable                  = true
+    required                 = false
+    string_attribute_constraints {
+      min_length = 1
+      max_length = 256
```

Comment:
The value `max_length` can be changed in the future with an update in place?

Reply:
Unfortunately the [AWS Cognito Doc](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html) says it is not possible to update an attributes once you add it to the user pool.

In this case, I'd use the max value possible for the `max_length`, which is `2048`.